src/Appwrite/Utopia/Response/Filters/V06.php

```diff
@@ -0,0 +1,288 @@
+                $parsedResponse = $this->parseCurrencyList($content);
+                break;
+
+            case Response::MODEL_ANY :
+                $parsedResponse = $content;
+                break;
+
+            default:
+                throw new Exception('Recevied invalid model : '.$model);
+        }
+
+        return $parsedResponse;
+    }
+
+    private function parseCollectionList(array $content)
+    {
+        foreach($content['collections'] as $key => $collection){
+            $content['collections'][$key] = $this->parseCollection($collection);
+        }
+        return $content;
+    }
+
+    private function parseCollection(array $content)
+    {
+        $content['$collection'] = Database::SYSTEM_COLLECTION_COLLECTIONS;
+        $content['structure'] = true;
+        return $content;
+    }
+
+    private function parseFile(array $content)
+    {
+        $content['$collection'] = Database::SYSTEM_COLLECTION_FILES;
+        $content['algorithm'] = 'gzip';
+        $content['comment'] = '';
+        $content['fileOpenSSLCipher'] = OpenSSL::CIPHER_AES_128_GCM;
+        $content['fileOpenSSLIV'] = '';
+        $content['fileOpenSSLTag'] = '';
+        $content['fileOpenSSLVersion'] = '';
+        $content['folderId'] = '';
+        $content['path'] = '';
+        $content['sizeActual'] = $content['sizeOriginal'];
+        $content['token'] = '';
+        return $content;
+    }
+
+    private function parseCurrencyList(array  $content) 
+    {
+        $content['locations'] = [];
+        $currencies = $content['currencies'];
+        $parsedResponse = [];
+        foreach($currencies as $currency) {
+            $currency['locations'] = [];
+            $parsedResponse[] = $currency;
+        }
+        $content['currencies'] = $parsedResponse;
+        return $content;
+    }
+
+    private function parseContinentList(array $content)
+    {
+        $continents = $content['continents'];
+        $parsedResponse = [];
+        foreach($continents as $continent) {
+            $parsedResponse[$continent['code']] = $continent['name'];
+        }
+        $content['continents'] = $parsedResponse;
+        return $content;
+    }
+
+    private function parsePhoneList(array $content)
+    {
+        $phones = $content['phones'];
+        $parsedResponse = [];
+        foreach($phones as $phone) {
+            $parsedResponse[$phone['countryCode']] = $phone['code'];
+        }
+        $content['phones'] = $parsedResponse;
+        return $content;
+    }
+
+    private function parseCountryList(array $content) 
+    {
+        $countries = $content['countries'];
+        $parsedResponse = [];
+        foreach($countries as $country) {
+            $parsedResponse[$country['code']] = $country['name'];
+        }
+        $content['countries'] = $parsedResponse;
+        return $content;
+    }
+
+    private function parseLocale(array $content) 
+    {
+        $content['ip'] = empty($content['ip']) ? '' : $content['ip'];
```

Comment:
Now it is nitpicking 😅 don't know if it has any impact on performance, but wouldn't this get the same result without a method call:

`$content['ip'] = $content['ip'] ? $content['ip'] : ''`

Reply:
`$content['ip'] = $content['ip'] ?? '';` can also do the trick

Reply:
Does `??` also takes care of an empty string and not just `null`?

Reply:
Yeah, totally right. Checking against an empty string would be stupid, since we make it an empty string anyway on the condition 😅